esphome/components/tuya/climate/tuya_climate.cpp

```diff
@@ -129,6 +178,106 @@ void TuyaClimate::control(const climate::ClimateCall &call) {
       ESP_LOGV(TAG, "Setting eco: %s", ONOFF(eco));
       this->parent_->set_boolean_datapoint_value(*this->eco_id_, eco);
     }
+    if (this->sleep_id_.has_value()) {
+      const bool sleep = preset == climate::CLIMATE_PRESET_SLEEP;
+      ESP_LOGV(TAG, "Setting sleep: %s", ONOFF(sleep));
+      this->parent_->set_boolean_datapoint_value(*this->sleep_id_, sleep);
+    }
+  }
+}
+
+void TuyaClimate::control_swing_mode_(const climate::ClimateCall &call) {
+  bool vertical_swing_changed = false;
+  bool horizontal_swing_changed = false;
+
+  if (call.get_swing_mode().has_value()) {
+    const auto swing_mode = *call.get_swing_mode();
+
+    switch (swing_mode) {
+      case climate::CLIMATE_SWING_OFF:
+        if (swing_vertical_ || swing_horizontal_) {
+          swing_vertical_ = false;
+          swing_horizontal_ = false;
```

Comment:
Member variables should always be prefixed with `this->` please.
```suggestion
          this->swing_vertical_ = false;
          this->swing_horizontal_ = false;
```

Reply:
changed. Thank you for this. Total oversight on my part.